src/Momento.Sdk/Config/Middleware/IMiddleware.cs

```diff
@@ -0,0 +1,23 @@
+using System.Threading.Tasks;
+using Momento.Sdk.Config.Retry;
+
+namespace Momento.Sdk.Config.Middleware;
+
+/// <summary>
+/// The Middleware interface allows the Configuration to provide a higher-order function that wraps all requests.
+/// This allows future support for things like client-side metrics or other diagnostics helpers.
+/// </summary>
+public interface IMiddleware
+{
+    public delegate Task<IGrpcResponse> MiddlewareFn(IGrpcRequest request);
```

Comment:
c# n00b question, what does `delegate` do?

Reply:
It's their way of typing functions to use functions as values/data

Reply:
roger ty